samples/ironside_se/wdtstart_sample/src/main.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/drivers/watchdog.h>
+
+static int wdt_channel_id = -1;
+static const struct device *wdt_dev;
+
+static void check_reset_cause(void)
+{
+	uint32_t reset_cause = 0;
+	int ret = hwinfo_get_reset_cause(&reset_cause);
+
+	if (ret != 0) {
+		printk("Failed to get reset cause: %d\n", ret);
+		return;
+	}
+
+	printk("Reset cause: 0x%08x\n", reset_cause);
+
+	if (reset_cause & RESET_WATCHDOG) {
+		printk("*** WATCHDOG RESET DETECTED ***\n");
+	} else if (reset_cause & RESET_PIN) {
+		printk("*** PIN RESET DETECTED ***\n");
+	} else if (reset_cause != 0) {
+		printk("Other reset cause detected\n");
+	} else {
+		printk("No reset detected (normal boot)\n");
+	}
+}
+
+static int init_watchdog(void)
+{
+	wdt_dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(wdt011));
+	if (wdt_dev == NULL) {
+		printk("Watchdog device not found\n");
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(wdt_dev)) {
+		printk("Watchdog device not ready\n");
+		return -ENODEV;
+	}
+
+	/* Configure timeout to match UICR.WDTSTART configuration */
+	struct wdt_timeout_cfg wdt_config = {
+		.window = {
+			.min = 0,
+			.max = 2000 /* 2 seconds in milliseconds */
+		},
+		.callback = NULL,
+		.flags = WDT_FLAG_RESET_SOC
+	};
+
+	wdt_channel_id = wdt_install_timeout(wdt_dev, &wdt_config);
+	if (wdt_channel_id < 0) {
+		printk("Failed to install watchdog timeout: %d\n", wdt_channel_id);
+		return wdt_channel_id;
+	}
+
+	/* Setup watchdog with minimal options */
+	int ret = wdt_setup(wdt_dev, 0);
+	if (ret != 0) {
+		printk("Failed to setup watchdog: %d\n", ret);
+		return ret;
+	}
+
+	printk("Watchdog driver initialized successfully\n");
+	return 0;
+}
+
+static void feed_watchdog(void)
+{
+	int ret = wdt_feed(wdt_dev, wdt_channel_id);
+
+	if (ret != 0) {
+		printk("Failed to feed watchdog: %d\n", ret);
+	}
+}
+
+int main(void)
+{
+	int err;
+	uint32_t reset_cause = 0;
+
+	printk("=== IronSide SE WDTSTART Sample ===\n\n");
+
+	/* Check reset cause using hwinfo API */
+	check_reset_cause();
+
+	err = hwinfo_get_reset_cause(&reset_cause);
+	if (err == 0 && (reset_cause & RESET_WATCHDOG)) {
+		/* Watchdog reset detected - continuously feed the watchdog */
+		printk("\nWatchdog reset detected - continuously feeding watchdog\n");
+		printk("This demonstrates that we can detect and handle watchdog resets\n\n");
+
+		/* Initialize watchdog driver */
+		err = init_watchdog();
```

Comment:
Maybe here or in the readme it could be nice to say re-initializing the watchdog is not a problem. This doesn't just initialize the driver it also configures the watchdog again, potentially with a different configuration.

Reply:
thanks, tiny typo now though :-)